src/api/topics.api.js

```diff
@@ -0,0 +1,212 @@
+     * @param {Layer[]} layersToActivate All layers that should be added to the displayed layer (but
+     *   not necessarily visible, that will depends on their state)
+     */
+    constructor(id, backgroundLayers, defaultBackgroundLayer, layersToActivate) {
+        this.id = id
+        this.backgroundLayers = [...backgroundLayers]
+        this.defaultBackgroundLayer = defaultBackgroundLayer
+        this.layersToActivate = [...layersToActivate]
+    }
+}
+
+/** @enum */
+export const topicTypes = {
+    THEME: 'THEME',
+    LAYER: 'LAYER',
+}
+
+/**
+ * Element of a topic tree, can be a node (or theme) or a leaf (a layer)
+ *
+ * @abstract
+ */
+class TopicTreeItem {
+    /**
+     * @param {Number | String} id The ID of the node
+     * @param {String} name The name of this node translated in the current lang
+     * @param {topicTypes} type The type of this node, layer or theme
+     */
+    constructor(id, name, type) {
+        this.id = id
+        this.name = name
+        this.type = type
+    }
+}
+
+/** Node of a topic three containing more themes and/or a list of layers */
+export class TopicTreeTheme extends TopicTreeItem {
+    /**
+     * @param {Number} id The ID of this node
+     * @param {String} name The name of the theme, in the current lang
+     * @param {TopicTreeItem[]} children All the children of this node (can be either layers or
+     *   themes, all mixed together)
+     */
+    constructor(id, name, children) {
+        super(id, name, topicTypes.THEME)
+        this.children = [...children]
+    }
+}
+
+/** A layer in the topic tree */
+export class TopicTreeLayer extends TopicTreeItem {
+    /**
+     * @param {String} layerId The BOD layer ID of this layer
+     * @param {String} name The name of this layer in the current lang
+     */
+    constructor(layerId, name) {
+        super(layerId, name, topicTypes.LAYER)
+        this.layerId = layerId
+    }
+}
+
+/**
+ * Reads the output of the topic tree endpoint, and creates all themes and layers object accordingly
+ *
+ * @param {TopicTreeItem} node The node for whom we are looking into
+ * @returns {TopicTreeItem}
+ */
+const readTopicTreeRecursive = (node) => {
+    if (node.category === 'topic') {
+        const children = []
+        node.children.forEach((topicChild) => {
+            children.push(readTopicTreeRecursive(topicChild))
+        })
+        return new TopicTreeTheme(node.id, node.label, children)
+    } else if (node.category === 'layer') {
+        return new TopicTreeLayer(node.layerBodId, node.label)
+    } else {
+        console.error('unknown topic node type', node.category)
+        return null
+    }
+}
+
+/**
+ * Loads the topic tree for a topic. This will be used to create the UI of the topic in the menu.
+ *
+ * @param {String} lang The lang in which to load the topic tree
+ * @param {Topic} topic The topic we want to load the topic tree
+ * @returns {TopicTreeItem[]} A list of topic tree nodes
```

Comment:
I would add `@async` and correct the returns to `{Promise<Array<TopicTreeItem|null>>}`